### DISCUSSION TITLE
useBalance return type is inconsistent

### DISCUSSION THREAD

**Discussion by CloudLooks** [wevm/wagmi#3471] [Q&A]
version 1.4.1

I'm noticing odd behavior in useBalance's return `data.value`
The arguments are `{ address, formatUnits: 'wei', token: undefined }`. The first few times this hook runs, the return value is a string. After a few render loops, the return value becomes a bigint as expected.

eg on Arbitrum, my balance is `0.05`.

```
const { data: balance } = useBalance({ address, formatUnits: "wei", token: isUsingEth ? undefined : currency });
console.log(balance.value); // "50000000000000000"
```

the same query starts returning the bigint representation of this value after a few renders `50000000000000000n`

**Comment by danaugrs** [comment]
I'm also experiencing this with the result of `useContractRead` hooks. Initially the value returned is an array of `string`s but later it becomes an array of `bigint`s.

**Comment by elee1766** [comment]
this is so funny

(can confirm that it happens with useBalance in 2.14.16)

**Reply by jxom** [reply]
https://github.com/wevm/wagmi/issues/new?template=bug_report.yml&title=useBalance%20return%20type%20is%20inconsistent
